src/format.unit.ts

```diff
@@ -88,4 +89,173 @@ describe("format", () => {
 		let form = await model.types.Form.create({}) as any;
 		expect(form.toString("[Table]")).toBe("Text2, Text2");
 	});
-});
\ No newline at end of file
+
```

Comment:
Added tests for Date/Number/Boolean custom formatting. For each field type the tests check that null or whitespace is parsed as null, and formatting and parsing for each field sub-type (i.e.: Date/Time, Integer/Decimal/Percentage). Also, the `convertBack` calls include leading and trailing whitespace to ensure that the formatters aren't affected by that.